linode/helper/provider_meta.go

```diff
@@ -0,0 +1,35 @@
+package helper
+
+import (
+	"fmt"
+
+	"github.com/linode/linodego"
+)
+
+func GetFwClientWithUserAgent(
+	resourceOrDataSourceUAComment string,
+	meta *FrameworkProviderMeta,
+) *linodego.Client {
+	client := meta.Client
+	client.SetUserAgent(generate(meta.ProviderUserAgent, resourceOrDataSourceUAComment))
+
+	return &client
+}
+
+func GetSDKClientWithUserAgent(
+	resourceOrDataSourceUAComment string,
```

Comment:
Just to simplify and unify these helpers a bit, would it make sense for them to accept the resource name and whether it's a resource or data source (enum?) as separate arguments?

Reply:
They takes different struct as parameters (our Framework provider metadata and SDKv2 provider metadata are separate structs), so I think it wouldn't look great if we combine them. What do you think?

Reply:
Good point, I think it makes sense to leave it as-is in that case 👍